### PR TITLE
Allow `/* ... */` comments in the VAST language

### PR DESCRIPTION
**Pull request by dominiklohmann**
This makes it possible to write comments in the VAST language directly using the `/* comment */` that is familiar from many programming languages.

### CODE REVIEW

File: vast/integration/vast_integration_suite.yaml [hunk 1]
```diff
@@ -533,9 +533,16 @@ tests:
...
-      - command: "export json 'pass'"
-      - command: "export json 'pass
-                               | select timestamp, flow_id, src_ip, dest_ip, src_port'"
+      - command: |
+          export json
+            'pass'
+      - command: |
+          export json
+            ' pass
+            /* a comment here */
```
Comment (Dakostu) [inline vast/integration/vast_integration_suite.yaml:542]: Do you think the comment functionality is mature enough to be documented on vast.io? E.g. in [pipelines/syntax](https://vast.io/docs/understand/language/pipelines#syntax) there is an example pipeline with step-by-step instructions afterwards. Maybe those instructions can become pipeline comments instead.

File: vast/integration/vast_integration_suite.yaml [hunk 2]
```diff
@@ -592,16 +599,22 @@ tests:
...
-                               | select timestamp, flow_id, src_ip, dest_ip, src_port
-                               | drop timestamp
-                               | hash --salt=\"abcdefghij12\" flow_id
-                               | drop flow_id
-                               | pseudonymize -m \"crypto-pan\" -s \"123456abcdef\" src_ip, dest_ip
-                               | extend abc=123,abc_str=\"123\", abc_list=[1,2,3]
-                               | rename new_list=abc_list
-                               | replace new_list=2
-                               | where #type ==\"suricata.alert\" || #type == \"suricata.fileinfo\"'"
+      - command: |
```
Comment (Dakostu) [inline vast/integration/vast_integration_suite.yaml:602]: Can there be an integration test that checks and confirms the intricacies of the multi-line comment syntax (i.e. invalid/valid syntax)?
e.g.
* `select timestamp | /**/`
* `select timestamp /*double ending*/ slash*/`
* `select timestamp /*double beginning /* is valid */`

**Review by Dakostu — APPROVED** [review]
I played around with the comment functionality - this gets an approval from me. Left some minor things.

File: vast/integration/vast_integration_suite.yaml (hunk 1 above)
Comment (dominiklohmann) [reply vast/integration/vast_integration_suite.yaml:542]: Good idea. I made this change in bb5fd5ebc.

File: vast/integration/vast_integration_suite.yaml [hunk 3]
```diff
@@ -533,9 +533,16 @@ tests:
...
-      - command: "export json 'pass
-                               | select timestamp, flow_id, src_ip, dest_ip, src_port'"
+      - command: |
+          export json
+            'pass'
+      - command: |
+          export json
+            ' pass
+            /* a comment here */
+            | select /* and a comemnt there /**/ timestamp, flow_id, src_ip, dest_ip, src_port
```
[the remark this answers is not in the document]
Comment (dominiklohmann) [reply vast/integration/vast_integration_suite.yaml]: Fixed in 7dde15904, thanks!

File: vast/integration/vast_integration_suite.yaml (hunk 2 above)
Comment (dominiklohmann) [reply vast/integration/vast_integration_suite.yaml:602]: Implemented exactly these three tests in d4c0e83fe.

**Pull request merged**